### PR TITLE
add ping node utils and update tests

### PR DESCRIPTION
**Pull request by A-Harby**
### Description

Add the Ping node function in utils to check every node that will be used in the tests before using it, preventing false-failing tests.

### CODE REVIEW

**Review by xmonader — CHANGES_REQUESTED** [review]
there's a pingNode function available on the client https://github.com/threefoldtech/tfgrid-sdk-ts/blob/e36762084aa34cc13a46c7141de0789ae244461f/packages/grid_client/src/modules/zos.ts#L46


P.S don't name functions `checkNodeAvail` it's incomplete and doesn't communicate the meaning

**Comment by A-Harby** [comment]
> there's a pingNode function available on the client
> 
> https://github.com/threefoldtech/tfgrid-sdk-ts/blob/e36762084aa34cc13a46c7141de0789ae244461f/packages/grid_client/src/modules/zos.ts#L46
> 
> P.S don't name functions `checkNodeAvail` it's incomplete and doesn't communicate the meaning

I'm using pingNode function to check node state and return its ID.
  
Updated function name https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1258/commits/30e3cb2a9dc9df986464eaa9c7c11b6719b59021

File: packages/grid_client/tests/utils.ts [hunk 1]
```diff
@@ -80,4 +81,24 @@ async function k8sWait(masterSSHClient, k8sMasterName, k8sWorkerName, waitTime,
...
+    try {
+      log(await gridClient.zos.pingNode({ nodeId: node.nodeId }));
+      break;
+    } catch (error) {
+      log("node " + node.nodeId + " is not responding, trying different node.");
+      Nodes.splice(+[Nodes.indexOf(node)], 1);
+    }
+  }
+  if (Nodes.length == 0) {
+    log("No nodes available with the needed resources for this test");
```
Comment (0oM4R) [inline packages/grid_client/tests/utils.ts:98]: is this function for test scripts only ?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Review by mohamedamer453 — APPROVED** [review]
Lgtm, and i ran the tests locally and everything went smoothly and passed.

image

**Pull request merged**